includes/gpio.h

```diff
@@ -0,0 +1,84 @@
+#ifndef GPIO_H
+#define GPIO_H
+
+#include <stdint.h>
+
+typedef struct 	GPIO_config_s
+{
+	uint32_t 	GPIO_MODER;  	/* GPIO mode register, 						Address offset 0x00 */
+	uint32_t 	GPIO_OTYPER; 	/* GPIO output type register, 				Address offset 0x04 */
+	uint32_t 	GPIO_OSPEEDR;	/* GPIO output speed register, 				Address offset 0x08 */
+	uint32_t 	GPIO_PUPDR;  	/* GPIO pull-up/pull-down register register,	Address offset 0x0c */
+	uint32_t 	GPIO_IDR;  		/* GPIO input data register, 					Address offset 0x10 */
+	uint32_t 	GPIO_ODR;  		/* GPIO output data register, 				Address offset 0x14 */
+	uint32_t 	GPIO_BSRR;  	/* GPIO bit set/reset register register, 		Address offset 0x18 */
+	uint32_t 	GPIO_LCKR;  	/* GPIO configuration lock register, 			Address offset 0x1c */
+}				GPIO_config;
```

Comment:
This is not exactly a config (BSRR is surely not). A better name would be a “GPIO port” or just “GPIO.”